apps/prairielearn/src/lib/externalGrader.ts

```diff
@@ -232,6 +234,11 @@ export async function processGradingResult(content: any): Promise<void> {
       content.grading.score,
       null, // `v2_score`: gross legacy, this can safely be null
     ]);
+
+    await updateCourseInstanceUsagesForGradingJob({
+      grading_job_id: content.gradingId,
+    });
```

Comment:
We had discussed offline that we'd like to avoid double-counting compute usage if a result message is processed twice, as would happen if a server dies in the middle of processing a result. This is already handled by the `grading_jobs_update_after_grading` sproc:

https://github.com/PrairieLearn/PrairieLearn/blob/a6bcc71785c5199eec9ce438df128feb9d400cb8/apps/prairielearn/src/sprocs/grading_jobs_update_after_grading.sql#L77-L81

Do you  still think we should do something about this?

Reply:
Good thinking. Fixed in 9d229d7214f389a3e26b188f6e323e7576191453

I opted to fetch the entire `grading_job` row to check whether `graded_at IS NULL`. This is currently a bit wasteful, but when we rewrite `grading_jobs_update_after_grading` in TS this will be the natural first step in any case.